### PR TITLE
Create And Refresh Pivot Table

### PR DESCRIPTION
**Pull request by openbots-aw**
- Added command for creating Pivot Table
- Added command for refreshing Pivot Table

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        [Description("Specify the Worksheet within the Workbook to extract Range and create Pivot Table.")]
+        [SampleUsage("Sheet1 || {vSheet}")]
+        [Remarks("")]
+        [Editor("ShowVariableHelper", typeof(UIAdditionalHelperType))]
+        public string v_SheetName { get; set; }
+
+        [Required]
+        [DisplayName("Source Table Name")]
+        [Description("Enter the Table name to extract data for Pivot Table.")]
+        [SampleUsage("Table|| {vTable}")]
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs:41]:
```suggestion
        [SampleUsage("Table || {vTable}")]
```

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            RenderedControls.AddRange(commandControls.CreateDefaultInputGroupFor("v_TableName", this, editor));
+            RenderedControls.AddRange(commandControls.CreateDefaultInputGroupFor("v_CellLocation", this, editor));
+            RenderedControls.AddRange(commandControls.CreateDefaultInputGroupFor("v_PivotTable", this, editor));
+
+            return RenderedControls;
+        }
+
+        public override string GetDisplayValue()
+        {
+            return base.GetDisplayValue() + $" [Create '{v_PivotTable}' - From '{v_TableName}' -Instance Name  '{v_InstanceName}']";
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs:121]:
```suggestion
            return base.GetDisplayValue() + $" [Create '{v_PivotTable}' From '{v_TableName}' - Instance Name  '{v_InstanceName}']";
```

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs [hunk 3]
```diff
@@ -0,0 +1,124 @@
+﻿using Microsoft.Office.Interop.Excel;
+using OpenBots.Core.Attributes.PropertyAttributes;
+using OpenBots.Core.Command;
+using OpenBots.Core.Enums;
+using OpenBots.Core.Infrastructure;
+using OpenBots.Core.Properties;
+using OpenBots.Core.Utilities.CommonUtilities;
+
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs:8]: These spaces were added in mass by mistake when I had to delete a namespace from the entire solution, but I generally try to remove them when I see them. Please don't include them when you copy/paste code to create new commands

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            var engine = (IAutomationEngineInstance)sender;
+            string vSheet = v_SheetName.ConvertUserVariableToString(engine);
+            var vTableName = v_TableName.ConvertUserVariableToString(engine);
+            var vCellLocation = v_CellLocation.ConvertUserVariableToString(engine);
+            var vPivotTable = v_PivotTable.ConvertUserVariableToString(engine);
+            var excelObject = v_InstanceName.GetAppInstance(engine);
+            var excelInstance = (Application)excelObject;
+            var workBook = excelInstance.ActiveWorkbook;
+            var workSheet = excelInstance.Sheets[vSheet] as Worksheet;
+            var excelTable = workSheet.ListObjects[vTableName];
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs:83]: This implies that a worksheet actually contains a worksheet.ListObjects to begin with, but I don't believe we have any commands that do that. Also, at first glance, it seemed to me like the input was asking for a studio DataTable variable. Please clarify this in the description

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs (hunk 4 above)
Comment (openbots-aw) [reply OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs:82]: worksheet.ListObjects is the list of Tables inside the excel sheet. And for creating Pivot Table sheet must contain a table from which the Pivot Table will be created. I have changed the name and description from Source Table Name to Excel Table Name.

File: OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs (hunk 3 above)
Comment (openbots-aw) [reply OpenBots.Commands/OpenBots.Commands.Microsoft/OpenBots.Commands.Excel/ExcelCreatePivotTableCommand.cs]: Done.

**Review by openbots-ch — APPROVED** [review]

**Review by openbots-ff — APPROVED** [review]

**Pull request merged**